### PR TITLE
fix: remove focus from myinfo child field when form is opened

### PR DESCRIPTION
**Pull request by wanlingt**
## Problem

When a form is opened, the focus immediately goes to the MyInfo child field. The focus should not be on any form field.

Closes FRM-1323

## Solution

To initialise the MyInfo child field, specify the focusOption `{shouldFocus: false}` in `append`. ([Docs here](https://react-hook-form.com/docs/usefieldarray)).

**Breaking Changes** 

- [x] No - this PR is backwards compatible  

## Before & After Screenshots

**BEFORE**:

https://github.com/opengovsg/FormSG/assets/56983748/7b6b2811-6319-4638-a630-e76583f0fcfe

**AFTER**:

https://github.com/opengovsg/FormSG/assets/56983748/c57ced59-cb46-40d8-926b-7d28a5cf792c

## Tests

1. MyInfo child field - 1 child only
- [ ] Open up a MyInfo child form with 1 child field
- [ ] The focus should not be on the child field, unless the field is clicked
- [ ] Select a child
- [ ] Submit the form
- [ ] Check that the email response contains the child's info

2. MyInfo child field - 2 children only (same test as above)
- [ ] Open up a MyInfo child form with 2 children fields
- [ ] The focus should not be on the child field, unless the field is clicked
- [ ] Select a child
- [ ] Submit the form
- [ ] Check that the email response contains the children's info

3. MyInfo child field - 1 child only with option to add children
- [ ] Enable the option "Allow respondent to add multiple children" on a MyInfo child form
- [ ] Open up a MyInfo child form with 1 child field
- [ ] The focus should not be on the child field, unless the field is clicked
- [ ] Select a child
- [ ] Add another child
- [ ] Select the 2nd child
- [ ] Submit the form
- [ ] Check that the email response contains the children's info

### CODE REVIEW

File: frontend/src/templates/Field/ChildrenCompound/ChildrenCompoundField.tsx [hunk 1]
```diff
@@ -94,9 +93,9 @@ export const ChildrenCompoundField = ({
   // Initialize with a single child section
   useEffect(() => {
     if (!fields || !fields.length) {
-      append([''])
+      update(0, '')
```
Comment (KenLSM) [inline frontend/src/templates/Field/ChildrenCompound/ChildrenCompoundField.tsx]: Saw that there's another parameter that we can set on append, would this work?

```suggestion
      append([''], { shouldFocus: false })
```

From the docs, update would do a re-render as it unmounts and remounts the component (`<ChildrenBody />`) which might be expensive.

Comment (wanlingt) [reply frontend/src/templates/Field/ChildrenCompound/ChildrenCompoundField.tsx]: ooh yeah, this works. thanks ken! where did you find the other parameters btw? I see `focusOptions` in the docs, but not what they are

Comment (KenLSM) [reply frontend/src/templates/Field/ChildrenCompound/ChildrenCompoundField.tsx]: Similarly, I tried to look at the docs, wasn't helpful. Then I looked into the typedocs for `append` and found it after running through a few rabbitholes 😆 .

**Review by KenLSM — APPROVED** [review]
LGTM! Thanks wanling!

**Pull request merged**